23.md

```diff
@@ -0,0 +1,91 @@
+NIP-23
+======
+
+Relays List
+-----------
+
+`draft` `optional` `author:fiatjaf` `author:cameri` `author:monlovesango` `author:giszmo`
+
+A special event with kind `10001`, meaning "relay list" is defined as having a list of tags, one for each relay the author uses.
+
+The content is not used.
+
+The tags consist of arrays of 3 elements: the first is the relay URL, the second is the _read_ condition, the third is the _write_ condition.
+
+The _read_ condition consists of a string containing a rule that follows a subset of the [runes](https://pypi.org/project/runes/) language. Specifically only the `|`, `&`, `!`, `>`, `<`, `/` and `=` operators are allowed. When the rule is an empty string it evaluates to `true`. All the operators must be tested against all possible values in the case of filter values that consist of lists of values and also in the case of event tags that can have multiple values for the same tag -- in other words, the `=` may be interpreted as a `values.any(v => v == runeValue)` instead of an `values == runeValue` operator; the `<` may be interpreted as a `values.any(v => v < runeValue)` and so on. Event tags are identified just by their tag key (for example, `e` or `p`) without the `#` prefix used in filters.
+
+The `read` rule operates on the values of the [NIP-01](01.md) **filter** object, while the `write` rule operates on the values of the **event** object.
+
+When a `read` rule evaluates to `true` for a given **filter** the client SHOULD send a `REQ` message to the relay, otherwise it SHOULD NOT. When a `write` rule evaluates to `true` for a given **event** the client SHOULD send that event (in a `EVENT` message) to the relay, otherwise it MUST NOT.
```

Comment:
Why be prescriptive in the `true` cases at all? Why "MUST NOT" in the false case for write? I see the "Don't re-send my DMs to any relay not compliant with nip-21" but then you also wouldn't want the client to REQ those events from a non-nip-21 relay? You would also not want to use more than one nip-21 relay for your DMs with any particular correspondence but would you want to put into your public event a list of your conspirators by relay?

```suggestion
When a `read` rule evaluates to `false` for a given **filter** the client SHOULD NOT send that filter in a `REQ` message to that relay.

When a `write` rule evaluates to `false` for a given **event** the client SHOULD NOT send that event in an `EVENT` message to that relay.
```

at the very least this typo and formatting:

```suggestion
When a `read` rule evaluates to `true` for a given **filter** the client SHOULD send that filter in a `REQ` message to the relay, otherwise it SHOULD NOT.

When a `write` rule evaluates to `true` for a given **event** the client SHOULD send that event in an `EVENT` message to the relay, otherwise it MUST NOT.
```